rs/rosetta-api/icp_ledger/ledger/src/main.rs

```diff
@@ -749,7 +752,39 @@ fn post_upgrade(args: Option<LedgerCanisterPayload>) {
     let start = dfn_core::api::performance_counter(0);
     let mut stable_reader = stable::StableReader::new();
     let mut ledger = LEDGER.write().unwrap();
-    *ledger = ciborium::de::from_reader(&mut stable_reader).expect("Decoding stable memory failed");
+    let mut pre_upgrade_instructions_consumed = 0;
+    match ciborium::de::from_reader(&mut stable_reader) {
```

Comment:
As discussed in https://github.com/dfinity/ic/pull/1448, we should use magic bytes to determine if memory manager is used.

Reply:
Done!